proton-j/src/main/java/org/apache/qpid/proton/engine/Event.java

```diff
@@ -80,15 +83,46 @@
         SELECTABLE_WRITABLE,
         SELECTABLE_EXPIRED,
         SELECTABLE_ERROR,
-        SELECTABLE_FINAL
+        SELECTABLE_FINAL,
+        
+        /**
+         * This value must never be used to generate an event, it's only used as
+         * a guard when casting custom EventTypes to builtin {@link Type} via
+         * {@link Event#getBuiltinType()}.
+         */
+        NOT_A_BUILTIN_TYPE;
```

Comment:
NOT_A_BUILTIN_TYPE similarly seems a bit off as a name. Perhaps following in the theme of the work, something like EXTENDED, or following the earlier suggestion, NON_CORE, etc?

Also, for the javadoc link, the Event#getBuiltinType()  method doesnt exist.